php/datasource/class-fieldmanager-datasource.php

```diff
@@ -7,66 +12,87 @@
  *
  * This class can be used as an arbitrary data source for static options, or it
  * can be extended to provide custom data sources.
- *
- * @package Fieldmanager_Datasource
  */
 class Fieldmanager_Datasource {
 
 	/**
+	 * The datasource options.
+	 *
 	 * @var array
 	 */
 	public $options = array();
 
 	/**
+	 * Options callback.
+	 *
 	 * @var boolean
 	 */
-	public $options_callback = Null;
+	public $options_callback = null;
 
 	/**
+	 * Whether or not to use AJAX.
+	 *
 	 * @var boolean
 	 */
-	public $use_ajax = False;
+	public $use_ajax = false;
 
 	/**
+	 * Allow option groups.
+	 *
 	 * @var boolean
 	 */
-	public $allow_optgroups = True;
+	public $allow_optgroups = true;
 
 	/**
+	 * The AJAX action.
+	 *
 	 * @var string
 	 */
 	public $ajax_action = '';
 
 	/**
-	 * @var int
 	 * Counter to create uniquely named AJAX actions.
+	 *
+	 * @var int
 	 */
 	public static $counter = 0;
 
 	/**
+	 * If true, group elements.
+	 *
 	 * @var boolean
-	 * If true, group elements
 	 */
-	public $grouped = False;
+	public $grouped = false;
 
 	/**
-	 * Constructor
+	 * Constructor.
+	 *
+	 * @throws FM_Developer_Exception If the property isn't public, don't set it (rare).
+	 *
+	 * @param array $options The datasource options.
 	 */
 	public function __construct( $options = array() ) {
 
 		foreach ( $options as $k => $v ) {
 			try {
-				$reflection = new ReflectionProperty( $this, $k ); // Would throw a ReflectionException if item doesn't exist (developer error)
-				if ( $reflection->isPublic() ) $this->$k = $v;
-				else throw new FM_Developer_Exception; // If the property isn't public, don't set it (rare)
+				// Would throw a ReflectionException if item doesn't exist (developer error).
+				$reflection = new ReflectionProperty( $this, $k );
+				if ( $reflection->isPublic() ) {
+					$this->$k = $v;
+				} else {
+					// If the property isn't public, don't set it (rare).
+					throw new FM_Developer_Exception();
+				}
 			} catch ( Exception $e ) {
 				$message = sprintf(
+					/* translators: 1: property name, 2: current class name, 3: option name */
 					__( 'You attempted to set a property "%1$s" that is nonexistant or invalid for an instance of "%2$s" named "%3$s".', 'fieldmanager' ),
-					$k, __CLASS__, !empty( $options['name'] ) ? $options['name'] : 'NULL'
+					$k,
+					__CLASS__,
```

Comment:
This class name is derived from `get_class()` earlier in the PR, and the translator comment is "field class": https://github.com/alleyinteractive/wordpress-fieldmanager/pull/645/files#diff-ae082fcb76024c82133d82bdef5c1930R434. Worth standardizing, etc.?